### ISSUE TITLE
Mismatch in exporting TPOT stacking estimator to PMML

### ISSUE THREAD

**Issue by nuttracy** [jpmml/sklearn2pmml#275]
Dear vruusmann,

I'm trying to build a stacking pipeline starts with logistic regression and followed with linear regression.
The pipeline uses tpot's StackingEstimator on logistic regression. 
However, when I output pmml using sklearn2pmml, I found out a mismatch on the columnTransformer. To be more specific, here is my code:

```
response_col = 2
base_estimator = LogisticRegression()
base_estimator.fit(X, Y)
yX1 = StackingEstimator(base_estimator).transform(X)
keep_score = ColumnTransformer([("keep", "passthrough", [response_col])], remainder = "drop")
y1 = keep_score.fit_transform(yX1)

meta_estimator = LinearRegression()
meta_estimator.coef_ = np.array([1.0]).T
meta_estimator.intercept_ = 0.0

pmml_pipeline = PMMLPipeline([
    ('pipeline_1', Pipeline([
        ('estimator', StackingEstimator(base_estimator)),
        ('keep_score', keep_score)])),
    ('meta_estimator', meta_estimator)
])
pmml_pipeline.target_fields = np.array(["PredProb"]) 
```
response_col is 2 because I want to use predict_proba[:,1] from the logistic regression. And from [tpot's stacking eistmator](https://github.com/EpistasisLab/tpot/blob/master/tpot/builtins/stacking_estimator.py), 0 column is predict, 1 column is predict_proba[:,0] and 2 is predict_proba[:,1]. 

But when I use sklearn2pmml to get the pmml file, and use pmml to score the data, I get prediction of class rather than the predict_proba. 
I've tested different value of response_col, it seems that if I want to get the predict_proba[:,1] from pmml, I need to change the response_col to 1. But then the pmml_pipeline has a different meaning.

Do you know how to handle this issue?

My sklearn version is 21.3, sklearn2pmml version is 48.0.

Thank you so much!

**Comment by vruusmann** [comment]
> And from tpot's stacking eistmator, 0 column is predict, 1 column is predict_proba[:,0] and 2 is predict_proba[:,1].

Looks like TPOT has rearranged its result matrix columns.

The predicted probabilities used to come before the predicted class label, now they come after.

**Comment by vruusmann** [comment]
> The predicted probabilities used to come before the predicted class label, now they come after.

There needs to be a configuration option for choosing which column layout to use.

**Comment by nuttracy** [comment]
> > The predicted probabilities used to come before the predicted class label, now they come after.
> 
> There needs to be a configuration option for choosing which column layout to use.

Thanks vruusmann. Do you think there is a way we can set the configuration in skearn2pmml?

**Comment by vruusmann** [comment]
> Do you think there is a way we can set the configuration in skearn2pmml?

My earlier resolution was wrong-ish - looks like the order of result matrix columns has been the same for as long as TPOT GitHub commit history shows.

I've fixed the JPMML-SkLearn library, and improved its TPOT integration test coverage. Will publish those code changes soon.

**Comment by nuttracy** [comment]
> > Do you think there is a way we can set the configuration in skearn2pmml?
> 
> My earlier resolution was wrong-ish - looks like the order of result matrix columns has been the same for as long as TPOT GitHub commit history shows.
> 
> I've fixed the JPMML-SkLearn library, and improved its TPOT integration test coverage. Will publish those code changes soon.

Sounds great. Thank you vruusmann!

**Comment by vruusmann** [comment]
@nuttracy What's the intent of your original pipeline? It's about extracting the probability of the "event" case as a `PredProb` field, and the `StackingEstomator` meta-estimator is just a clever workaround to achieve that?

If so, then your original pipeline could be simplified as follows:
```python
from sklearn2pmml.decoration import Alias
from sklearn2pmml.preprocessing import ExpressionTransformer

pipeline = PMMLPipeline([
  ("clasifier", LogisticRegression())
], predict_proba_transformer = Alias(ExpressionTransformer("X[1]"), "PredProba", prefit = True))
pipeline.fit(X, y)

sklearn2pmml(pipeline, "PipelineWithPredProba.pmml")
```

The highlighted functionality is the `PMMLPipeline.predict_proba_transformer` attribute.

**Comment by nuttracy** [comment]
> @nuttracy What's the intent of your original pipeline? It's about extracting the probability of the "event" case as a `PredProb` field, and the `StackingEstomator` meta-estimator is just a clever workaround to achieve that?
> 
> If so, then your original pipeline could be simplified as follows:
> 
> ```python
> from sklearn2pmml.decoration import Alias
> from sklearn2pmml.preprocessing import ExpressionTransformer
> 
> pipeline = PMMLPipeline([
>   ("clasifier", LogisticRegression())
> ], predict_proba_transformer = Alias(ExpressionTransformer("X[1]"), "PredProba", prefit = True))
> pipeline.fit(X, y)
> 
> sklearn2pmml(pipeline, "PipelineWithPredProba.pmml")
> ```
> 
> The highlighted functionality is the `PMMLPipeline.predict_proba_transformer` attribute.

Hi vruusmann, the original pipeline is a simplified version of what I want to do -- to help me to identify the mismatch issue. Basically I want to build a stacking pipeline using two classifiers and the final prediction is the difference of the predict_proba of these two. So I used a FeatureUnion to union two classifiers and linear regression to get the difference.
Here is the code.
```
pmml_pipeline = PMMLPipeline([
    ('stacking_estimators_output',FeatureUnion([
        ('base_pipeline_t', Pipeline([
            ("base_feature_filtering_t", base_feature_filtering1),
            ("base_estimator_t", StackingEstimator(base_estimator1)),
            ('keep_score1', keep_score1)
        ])),
        ('base_pipeline_c', Pipeline([
            ("base_feature_filtering_c", base_feature_filtering2),
            ("base_estimator_c", StackingEstimator(base_estimator2)),
            ('keep_score2', keep_score2)
        ]))
    ])),
    ('meta_estimator', meta_estimator)
])
```

Thanks!
